### PR TITLE
Include Timestamps with ISO 8601 format

### PR DESCRIPTION
**Pull request by karupayun**
In this PR we are adding the possibility to choose ISO-8601 format to
display time. This will be used in a following PR to display labels in
the Timeline.

The details of the implementation can be found in the doc:
go/stadia-orbit-timeline-improvements

Bug: http://b/162485497

Test: Unit tests, but also tested in a real capture (which will be
connected in a following PR).

### CODE REVIEW

**Review by pierricgimmig — COMMENTED** [review]
Hi Pablo, thanks for this, please see comments.

File: src/DisplayFormats/DisplayFormats.cpp [hunk 1]
```diff
@@ -45,4 +45,50 @@ std::string GetDisplayTime(absl::Duration duration) {
...
 
+std::string ToStringAtLeastTwoDigits(int number) {
+  return std::to_string(number / 10) + std::to_string(number % 10);
+}
+
+std::string GetDisplayISOTimestamp(absl::Duration timestamp, int num_digits_precision,
+                                   absl::Duration total_capture_duration) {
+  std::string label;
+  // Hours, minutes and seconds
+  if (total_capture_duration >= absl::Hours(1)) {
```
[the remark this answers is not in the document]
Comment (karupayun) [reply src/DisplayFormats/DisplayFormats.cpp:56]: Yes 100% intended. I'm out of ideas how to include capture_duration here to make it easier to understand. 

Basically, if the capture is longer than a minute, for consistency we will show the minute part in all the timestamps, so we will have 00:01 to represent 1 second. Similar to the hour part, if the capture is longer than 1 hour, we will always show the hour part. This was discussed with Carsten but we can discuss it again on the weekly UI meeting. 

Feel free to suggest a better way to implement it. Maybe an enum to represent the 3 values? (short than a minute, short than an hour, longer than an hour). I don't like it either.

File: src/DisplayFormats/DisplayFormatsTest.cpp [hunk 2]
```diff
@@ -27,4 +27,33 @@ TEST(DisplayFormats, GetDisplayTime) {
   EXPECT_EQ(GetDisplayTime(absl::Nanoseconds(24 * 60 * 60 * 12'345'600'000ULL)), "12.346 days");
 }
 
+TEST(DisplayFormats, GetDisplayISOTimestamp) {
```
Comment (reichlfl) [inline src/DisplayFormats/DisplayFormatsTest.cpp:30]: You could add a test where an "intermediate" unit is missing and expected to be filled to "00" (for example "01:00.123"). You also don't have a single test that includes an hour timestamp, maybe add one?

**Review by reichlfl — APPROVED** [review]

File: src/DisplayFormats/DisplayFormatsTest.cpp (hunk 2 above)
Comment (karupayun) [reply src/DisplayFormats/DisplayFormatsTest.cpp:30]: perfect. Replaced two non-very useful tests with "1:00.234" (1 minute and something) and "01:00:00.00000" (1 hour with 5 precision-digits)

File: src/DisplayFormats/DisplayFormats.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (pierricgimmig) [reply src/DisplayFormats/DisplayFormats.cpp:56]: Ok sounds good, simply wanted to double check, if this is intended behavior, fine with me. Thanks for the info.

**Review by pierricgimmig — APPROVED** [review]

**Pull request merged**